workers/loc.api/sync/helpers/calc-grouped-data.js

```diff
@@ -59,18 +75,32 @@ const _mergeData = (data) => {
         return
       }
 
-      res.forEach((item, index) => {
+      for (const [index, item] of iterator.entries()) {
         if (mts === item.mts) {
           res[index] = {
             ...item,
             [key]: { ...vals }
           }
+
+          return
         }
-      })
+        if (
+          res[index + 1] &&
+          res[index + 1].mts < mts &&
+          res[index].mts > mts
+        ) {
+          res.splice(index + 1, 0, {
+            mts,
+            [key]: { ...vals }
```

Comment:
if looking for speed here we should use Object.assign check the following performance comparison: https://frontbackend.com/javascript/what-is-the-the-fastest-way-to-deep-clone-an-object-in-javascript#:~:text=According%20to%20the%20benchmark%20test,assign%20supports%20only%20shallow%20copy.
Same on other places where spread syntax is used as to clone the objects.